src/invariant/ConstantProductInvariant.cpp

```diff
@@ -0,0 +1,96 @@
+#ifdef BUILD_TESTS
```

Comment:
missing Visual Studio project changes; Also, should this just be in the `src/invariant/test` folder as it's test only?